moped-editor/src/views/projects/projectView/KnackSync.js

```diff
@@ -0,0 +1,217 @@
+import React from "react";
+import { MenuItem, Icon, ListItemIcon, ListItemText } from "@material-ui/core";
+
+import { useMutation } from "@apollo/client";
+
+import { UPDATE_PROJECT_KNACK_ID } from "../../../queries/project";
+
+/**
+ * Entry in project menu drop down to trigger sync to Data Tracker
+ * @return {JSX.Element}
+ * @constructor
+ */
+const KnackSync = React.forwardRef(
+  (
+    {
+      project,
+      closeHandler,
+      snackbarHandler,
+      sceneNumber = 514,
+      viewNumber = 3047,
+      knackApplicationId = "6167314778435d001ea3e7cb",
+    },
+    ref
+  ) => {
```

Comment:
This feels like an anti-pattern:
1. The default values should probably be part of an environment variable, since I suspect we are going to have multiple stages to test this code with.
2. React.forwardRef probably doesn't belong there for this large component, I don't see ref being used or passed explicitly within this code, though I suspect MenuItem may require it. 
3. We should consider going the custom hook route for real reusability.

Reply:
i agree. @frankhereford unless i'm missing something you can just drop the forwardRef and declare the component with 
```javascript
const KnackSync = ({
      project,
      closeHandler,
      snackbarHandler,
    }
  ) => { ... }
```

Reply:
At a glance it looks like you could do away with the `project.moped_project[0]` accessor by passing the value of it as the `project` pop. I definitely prefer that vs repeating `project.moped_project[0]` throughout.

Reply:
This is very helpful input - thank you both. 

I totally agree that the values that define the knack endpoint should be pulled out of the app, and I've moved them into env variables. 👍 

I also think that only handling a slice of the hasura response down as a property to the component cleans up a bunch of repetitious and ugly code, so that's a ✅ too.

Re the React.forwardRef, initially was using the vanilla function form like John suggested, but at some point, I started getting this error. 

![image](https://user-images.githubusercontent.com/3653206/142272750-950f6b93-9796-4495-a341-31dcb4e746df.png)

I kinda blindly followed the advice it offered, and I managed to get a state where I was not introducing another error in the console, so I went the forwardRef route.

That said, I don't fully understand why it's correct, and I'd be hard pressed to justify the choice. Based on @sergiogcx's comment, I suspect he is on the right track because this component needs a way to reach up a level in the hierarchy to close the menu. I'd be eager to read up or chat with y'all about what is the real answer here for long term maintainability and reuse.